app/controllers/packages_controller.rb

```diff
@@ -0,0 +1,41 @@
+class PackagesController < ApplicationController
+  before_action :authenticate_user!, except: %i[index show]
+  load_and_authorize_resource except: %i[index show]
+
+  respond_to :json
+
+  def index
+    @packages = Package.includes(:user)
```

Comment:
[OPTIONAL] instance variables are not necessary if you are not passing to views. Can simply use `packages` like so: `packages = Package.includes(:user)`

Reply:
See where else this is applicable in your controllers

Reply:
Thank you very much @mcrd25 for your all your feedback, we will address them